File: source-connector/src/main/java/sourceconnector/domain/processor/impl/EmptyFilterProcessor.java
package sourceconnector.domain.processor.impl;

import sourceconnector.domain.log.Log;
import sourceconnector.domain.processor.AbstractFilterProcessor;

/**
 * Filter out the empty Log
 */
public class EmptyFilterProcessor extends AbstractFilterProcessor<Log> {
  @Override
  protected boolean condition(Log input) {
    return !input.get().isBlank();
  }
}

File: source-connector/src/main/kotlin/sourceconnector/domain/processor/impl/EmptyFilterProcessor.kt
package sourceconnector.domain.processor.impl

import sourceconnector.domain.log.Log
import sourceconnector.domain.processor.AbstractFilterProcessor

/**
 * Filter out the empty Log
 */
class EmptyFilterProcessor : AbstractFilterProcessor<Log>() {
  override fun condition(input: Log): Boolean {
    return input.get().isNotBlank()
  }
}
